File: static/js/rps.js
const RPS = {
    userScore: 0,
    computerScore: 0,
    emojis: { rock: '🪨', paper: '📄', scissors: '✂️' },

    render: function() {
        const div = document.createElement('div');
        div.innerHTML = `
            <h2 style="color:var(--accent);margin-bottom:1.5rem;">Rock Paper Scissors</h2>
            <div class="flex-center" style="margin-bottom:2rem;">
                <div class="score-display" style="flex:1;min-width:120px;">
                    <h3>You</h3>
                    <p id="userScore">0</p>
                </div>
                <div style="font-size:2rem;color:var(--text-secondary);">VS</div>
                <div class="score-display" style="flex:1;min-width:120px;">
                    <h3>Computer</h3>
                    <p id="compScore">0</p>
                </div>
            </div>
            <div class="flex-center" style="margin-bottom:2rem;">
                <button onclick="RPS.play('rock')" style="font-size:2rem;padding:1rem 2rem;background:var(--bg-primary);border:2px solid var(--accent);">🪨<br><small>Rock</small></button>
                <button onclick="RPS.play('paper')" style="font-size:2rem;padding:1rem 2rem;background:var(--bg-primary);border:2px solid var(--accent);">📄<br><small>Paper</small></button>
                <button onclick="RPS.play('scissors')" style="font-size:2rem;padding:1rem 2rem;background:var(--bg-primary);border:2px solid var(--accent);">✂️<br><small>Scissors</small></button>
            </div>
            <div id="rpsResult" style="text-align:center;font-size:1.5rem;min-height:80px;padding:1rem;background:var(--bg-primary);border-radius:var(--border-radius);margin-bottom:1rem;"></div>
            <div class="flex-center">
                <button class="danger" onclick="RPS.reset()">Reset Game</button>
            </div>
        `;
        return div;
    },

    init: function() {
        const saved = Storage.load('app_rps_scores', { userScore: 0, computerScore: 0 });
        this.userScore = saved.userScore;
        this.computerScore = saved.computerScore;
        this.updateScores();
    },

    play: function(choice) {
        const choices = ['rock', 'paper', 'scissors'];
        const comp = choices[Math.floor(Math.random() * 3)];

        let result;
        if (choice === comp) result = 'tie';
        else if (
            (choice === 'rock'     && comp === 'scissors') ||
            (choice === 'paper'    && comp === 'rock')     ||
            (choice === 'scissors' && comp === 'paper')
        ) result = 'win';
        else result = 'lose';

        if (result === 'win') this.userScore++;
        else if (result === 'lose') this.computerScore++;

        this.updateScores();
        this.showResult(choice, comp, result);
        Storage.save('app_rps_scores', { userScore: this.userScore, computerScore: this.computerScore });
    },

    showResult: function(userChoice, compChoice, result) {
        const el = document.getElementById('rpsResult');
        if (!el) return;
        const colors = { win: 'var(--success)', lose: 'var(--error)', tie: 'var(--warning)' };
        const labels = { win: 'You Win! 🎉', lose: 'You Lose! 😞', tie: "It's a Tie! 🤝" };
        el.innerHTML = `
            <div>${this.emojis[userChoice]} You &nbsp;&nbsp; vs &nbsp;&nbsp; Computer ${this.emojis[compChoice]}</div>
            <div style="color:${colors[result]};font-weight:bold;margin-top:0.5rem;">${labels[result]}</div>
        `;
    },

    reset: function() {
        if (!confirm('Reset game scores?')) return;
        this.userScore = 0;
        this.computerScore = 0;
        this.updateScores();
        Storage.save('app_rps_scores', { userScore: 0, computerScore: 0 });
        const el = document.getElementById('rpsResult');
        if (el) el.innerHTML = '';
    },

    updateScores: function() {
        const u = document.getElementById('userScore');
        const c = document.getElementById('compScore');
        if (u) u.textContent = this.userScore;
        if (c) c.textContent = this.computerScore;
    }
};

File: app/modules/rps.py
from flask import Blueprint, request, jsonify
import random

rps_bp = Blueprint('rps', __name__, url_prefix='/api/rps')

# Game state
game_state = {
    'user_score': 0,
    'computer_score': 0
}

def get_computer_choice():
    """Get random computer choice"""
    return random.choice(['rock', 'paper', 'scissors'])

def determine_winner(user_choice, computer_choice):
    """Determine winner of round"""
    user_choice = user_choice.lower()
    computer_choice = computer_choice.lower()
    
    if user_choice == computer_choice:
        return 'tie'
    
    if user_choice == 'rock' and computer_choice == 'scissors':
        return 'win'
    elif user_choice == 'paper' and computer_choice == 'rock':
        return 'win'
    elif user_choice == 'scissors' and computer_choice == 'paper':
        return 'win'
    else:
        return 'lose'

@rps_bp.route('/play', methods=['POST'])
def play():
    """Play one round"""
    data = request.get_json()
    user_choice = data.get('choice', '').lower()
    
    if user_choice not in ['rock', 'paper', 'scissors']:
        return jsonify({'error': 'Invalid choice'}), 400
    
    computer_choice = get_computer_choice()
    result = determine_winner(user_choice, computer_choice)
    
    if result == 'win':
        game_state['user_score'] += 1
    elif result == 'lose':
        game_state['computer_score'] += 1
    
    return jsonify({
        'user_choice': user_choice,
        'computer_choice': computer_choice,
        'result': result,
        'user_score': game_state['user_score'],
        'computer_score': game_state['computer_score']
    })

@rps_bp.route('/scores', methods=['GET'])
def get_scores():
    """Get current scores"""
    return jsonify(game_state)

@rps_bp.route('/reset', methods=['POST'])
def reset():
    """Reset scores"""
    game_state['user_score'] = 0
    game_state['computer_score'] = 0
    return jsonify(game_state)
